src/sqlitestorage.cpp

```diff
@@ -1848,7 +1848,19 @@ void SqliteStorage::calendarModified(bool modified, Calendar *calendar)
 
 void SqliteStorage::calendarIncidenceAdded(const Incidence::Ptr &incidence)
 {
-    if (!d->mIncidencesToInsert.contains(incidence->uid(), incidence) && !d->mIsLoading) {
+    QMultiHash<QString, KCalendarCore::Incidence::Ptr>::Iterator deleted =
+        d->mIncidencesToDelete.find(incidence->uid());
+    if (deleted != d->mIncidencesToDelete.end() && !d->mIsLoading) {
+        qCDebug(lcMkcal) << "removing incidence from deleted" << incidence->uid();
+        while (deleted != d->mIncidencesToDelete.end()) {
+            if ((*deleted)->recurrenceId() == incidence->recurrenceId()) {
+                deleted = d->mIncidencesToDelete.erase(deleted);
+                calendarIncidenceChanged(incidence);
+            } else {
+                ++deleted;
+            }
+        }
+    } else if (!d->mIncidencesToInsert.contains(incidence->uid(), incidence) && !d->mIsLoading) {
```

Comment:
might be best to now hoist this d->mIsLoading check up, so that we early-out return prior to initializing the `deleted` variable.

Reply:
Sure, that's right. Updated.